torch/_dynamo/debug_utils.py

```diff
@@ -291,16 +294,20 @@ def clone_inputs_retaining_gradness(example_inputs):
     return cloned_inputs
 
 
-def run_fwd_maybe_bwd(gm, args, only_fwd=False):
+def run_fwd_maybe_bwd(gm, args, only_fwd=False, disable_clone=False):
     """
     Runs a forward and possibly backward iteration for a given mod and args.
+
+    When disable_clone is True, we will use args as-is without cloning.
+    This is higher fidelity but we may destroy the args in the process.
```

Comment:
Is that in the case where the gm performs any inplace operation on the inputs? Or something else?
Could we clarify when they get destroyed?

Reply:
Yes, gm may mutate the inputs, thereby "destroying" them.